atc/creds/vault/vault_test.go

```diff
@@ -274,3 +297,687 @@ var _ = Describe("Vault", func() {
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/concourse/data/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, createMockV2Secret("foo")),
+					),
+				)
+				_, found, err := variables.Get(varFoo)
+				Expect(found).To(BeFalse())
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("allowRootPath", func() {
+			BeforeEach(func() {
+				server.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/sys/internal/ui/mounts/concourse/team/pipeline/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, &returnedMountInfo),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/concourse/data/team/pipeline/foo"),
+						ghttp.RespondWith(404, ""),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/sys/internal/ui/mounts/concourse/team/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, &returnedMountInfo),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/concourse/data/team/foo"),
+						ghttp.RespondWith(404, ""),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/sys/internal/ui/mounts/concourse/shared/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, &returnedMountInfo),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/concourse/data/shared/foo"),
+						ghttp.RespondWith(404, ""),
+					),
+					// These 2 should only be called for root.
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/sys/internal/ui/mounts/concourse/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, &returnedMountInfo),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v1/concourse/data/foo"),
+						ghttp.RespondWithJSONEncodedPtr(&statusCodeOK, createMockV2Secret("foo")),
+					),
+				)
+			})
+
+			Context("is true", func() {
+				BeforeEach(func() {
+					variables = creds.NewVariables(v, "team", "pipeline", true)
+				})
+
+				It("should get secret from root", func() {
+					_, found, err := variables.Get(varFoo)
+					Expect(err).To(BeNil())
+					Expect(found).To(BeTrue())
+				})
+			})
+
+			Context("is false", func() {
+				BeforeEach(func() {
+					variables = creds.NewVariables(v, "team", "pipeline", false)
+				})
+
+				It("should not get secret from root", func() {
+					_, found, err := variables.Get(varFoo)
+					Expect(err).To(BeNil())
+					Expect(found).To(BeFalse())
+				})
+			})
+		})
+	})
+})
+
+var _ = Describe("Vault KV1", func() {
+
+	var server *ghttp.Server
+	var returnedMountInfo vaultapi.Secret
+	var statusCodeOK int
+
+	var v *vault.Vault
+	var variables vars.Variables
+	var vaultApi *vault.APIClient
+	var varFoo vars.Reference
+
+	BeforeEach(func() {
+		server = ghttp.NewServer()
+
+		var err error
+		vaultApi, err = vault.NewAPIClient(lagertest.NewTestLogger("test"), "http://"+server.Addr(), vault.TLSConfig{}, vault.AuthConfig{}, "")
```

Comment:
to catch up with `master`:

```suggestion
		vaultApi, err = vault.NewAPIClient(lagertest.NewTestLogger("test"), "http://"+server.Addr(), vault.TLSConfig{}, vault.AuthConfig{}, "", 0)
```